src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -203,6 +204,8 @@ struct task_metrics {
   // The amount of time that this thread has lost due to retries (not including blocked time)
   long time_lost_nanos = 0;
 
+  long max_memory_allocated = 0;
```

Comment:
Again because both GPU and CPU memory is processed by this file can we please mark this as gpu_max_memory_allocated?